cmd/minishift/cmd/root.go

```diff
@@ -213,14 +222,37 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If `minishift profile set` or `minishift start --profile` command is used.
 	for i, arg := range os.Args {
+		// This will match if `--profile` flag is used
 		if arg == "--"+profileFlag {
+			// This make sure if user specify `--profile` flag without any name
+			// it should not panic with out of index error
+			if len(os.Args) <= i+1 {
+				break
+			}
 			profileName = os.Args[i+1]
 			if !cmdUtil.IsValidProfileName(profileName) {
 				atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
 			}
 			break
 		}
+		// This will match if we used profile or it's alias commands
+		if arg == profileFlag || arg == profileAlias[0] || arg == profileAlias[1] {
+			// This make sure if user specify profile command without any subcommand then
+			// it should not panic with out of index error
+			if len(os.Args) <= i+2 {
+				break
+			}
```

Comment:
We do not need this check as Cobra/Viper takes care of this. In my testing I found this as dead code.

Reply:
@LalatenduMohanty  this is not dead code. Try to comment it out and then do `minishift profile` you will get out of index runtime panic.

Reply:
@praveenkumar you are right. This check is required in this place.